client/components/Cart.js

```diff
@@ -0,0 +1,71 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import axios from "axios";
+import { fetchCart, deleteFromCart } from "../store/cart";
+
+class Cart extends Component {
+  constructor(props) {
+    super(props);
+    //this.deleteCartItem = this.deleteCartItem.bind(this);
+  }
+
+  async componentDidMount() {
+    this.props.fetchCart();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.products !== prevProps.products) {
+      this.props.fetchCart();
+    }
+  }
+
+  // deleteCartItem(lineId, lineQuantity) {
+  //   this.props.deleteLineitem(lineId, lineQuantity);
+  // }
+
+  render() {
+    const { cart } = this.props;
+    console.log('cart', cart);
+    return (
+      <div>
+        <h2>Cart:</h2>
+        {cart.map((item) => {
+          return (
+            <div key={item.id}>
+              <ul>
+                {
+                item.lineitems.map((line) => {
+                  return (
+                    <li key={line.id}>
+                      {line.product?.name}({line.quantity})
+                      <button
+                        className="delete"
+                        type="delete"
+                        onClick={() =>
+                          this.props.deleteLineitem(line.id, line.quantity)
+                        }
+                      >
+                        {" "}
+                        Delete{" "}
```

Comment:
in this component, in the future, you'll probably display a "total price" of everything in the cart
BE CAREFUL WHEN DOING THAT
what is `1.3 * 1.4?` irl it's 1.82 but in javascript it's `1.8199999999999998` and you really dont want to display `1.8199999999999998` on the screen!
the way you fix it is just... `.toFixed(2)`
```js
const num = 1.3 * 1.4;
console.log(num); /// 1.8199999999999998
console.log(num.toFixed(2)); // 1.82
```
this is called a "floating point precision error"